### PR TITLE
bump up maven-release-plugin version for JDK17

### PR DESCRIPTION
**Pull request by yoikawa**
trying to fix `maven-release-plugin` [failure](https://github.com/salesforce/grammaticus/actions/runs/4849990583/jobs/8642494462).

confirmed locally by running `mvn release:prepare -DdryRun`. let's see if this works.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,11 +31,11 @@
         <organization>Salesforce</organization>
     </developer>
     <developer>
-        <name>Steven Tamm</name>
+        <name>Andrzej Poblocki</name>
         <roles>
             <role>Member</role>
         </roles>
-        <email>stamm@salesforce.com </email>
+        <email>apoblock@salesforce.com </email>
```
Comment (apoblock) [inline pom.xml:38]: @yoikawa, I don't feel I deserve any credit here, but my email is `apoblocki@salesforce.com`

File: pom.xml [hunk 2]
```diff
@@ -31,26 +31,17 @@
...
-        <email>bo.yang@salesforce.com</email>
+        <email>apoblocki@salesforce.com</email>
         <organization>Salesforce</organization>
     </developer>
   </developers>
 
   <properties>
-    <maven.compiler.source>17</maven.compiler.source>
-    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.release>11</maven.compiler.release>
```
Comment (yoikawa) [inline pom.xml:44]: The [`release` property](https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#release) should be good enough to build project by targeting Java 11.   Tested w/ JDK11 and 17 compiler to build this project

File: pom.xml [hunk 3]
```diff
@@ -113,7 +104,7 @@
                     the BSD 3-Clause license. For full license text, see the LICENSE.txt file in the
                     repository.
                 </bottom>
-                <additionalparam>-Xdoclint:none</additionalparam>
+                <doclint>none</doclint>
```
Comment (yoikawa) [inline pom.xml:107]: The `maven-javadoc-plugin` 3.x requires `<doclint>` instead of `<additionalparam>`

File: pom.xml [hunk 4]
```diff
@@ -77,6 +68,10 @@
       <!-- Nexus Staging -->
       <build>
         <plugins>
+          <plugin>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.11.0</version>
+          </plugin>
```
Comment (yoikawa) [inline pom.xml:74]: the default version was 3.1.  `<maven.compiler.release>` requires 3.6 or later..

**Review by apoblock — APPROVED** [review]

**Pull request merged**